### DISCUSSION TITLE
HTTPStatusError does not show the HTTP method

### DISCUSSION THREAD

**Discussion by wmalik** [encode/httpx#1976] [Ideas]
The HTTPStatusError raised by the `raise_for_status` method contains the URL but not the HTTP method in the message. It would be nice to have the HTTP method too. I would be happy to submit a PR for this.

### Current format
```
HTTPStatusError: Client error '404 Not Found' for url 'https://someapi.com/somepath'
```

### Proposed format
```
HTTPStatusError: Client error '404 Not Found' for 'POST https://someapi.com/somepath'
```

**Comment by lovelydinosaur** [comment]
Interesting.
Did you hit this requirement because of a particular API or service that 404s for some method types but not others?

I'm not sure if I'm in favour of this or not, but I suppose if we did want it then perhaps this'd be appropriate?...

`HTTPStatusError: Client error '404 Not Found' for 'POST' request to 'https://someapi.com/somepath'`

**Comment by wmalik** [comment]
@tomchristie I recently started using httpx and just happened to notice the missing HTTP method, so this is purely a nice-to-have feature request. However, I can imagine the following scenario where this could be useful:

* A client that makes a `PATCH` request followed by a `GET` request to the same URL (in that case, it would be convenient to see in logs which of those requests produced a 404)

What do you think?

The message format you proposed sounds good to me.

**Reply by lovelydinosaur** [reply]
I'm not sure I mind much either way.
If anyone else here has a preference in either direction then I'd be okay to go with that I guess?
